cylc/flow/config.py

```diff
@@ -1179,16 +1179,12 @@ def compute_family_tree(self):
         for name in self.cfg['runtime']:
             ancestors = self.runtime['linearized ancestors'][name]
             for p in ancestors[1:]:
-                if p not in self.runtime['descendants']:
-                    self.runtime['descendants'][p] = []
-                if name not in self.runtime['descendants'][p]:
-                    self.runtime['descendants'][p].append(name)
+                self.runtime['descendants'].setdefault(p, set()).add(name)
```

Comment:
I probably spent too much time looking into this... but speedup looks impressive : https://github.com/wxtim/Notes/blob/master/Olivers_opt.ipynb